src/include/stir/Scanner.inl

```diff
@@ -154,25 +164,28 @@ Scanner::get_num_axial_blocks() const
 {
   // when using virtual crystals between blocks, there won't be one at the end, so we
   // need to take this into account.
-  return (num_rings+get_num_virtual_axial_crystals_per_block())/num_axial_crystals_per_block;
+  return (num_rings)/num_axial_crystals_per_block;
 }
 
 int
 Scanner::get_num_transaxial_blocks() const
 {
-  return num_detectors_per_ring/num_transaxial_crystals_per_block;
+  return num_transaxial_crystals_per_block == 0 ? num_detectors_per_ring :
```

Comment:
`num_transaxial_crystals_per_block` should never be zero. It doesn't make sense in STIR. You might have only 1 block, but no crystals in a block means no crystals